catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/suggestion/DmsCoordinateProcessor.java

```diff
@@ -0,0 +1,215 @@
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.ui.query.suggestion;
+
+import java.text.DecimalFormat;
+import java.text.NumberFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.annotation.Nullable;
+import org.codice.ddf.spatial.geocoding.Suggestion;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A regex-based processor for identifying potential DMS coordinate literals in a given input string
+ * despite deviations from the official spec. The general policy is to be as forgiving as possible
+ * without sacrificing determinism.
+ */
+public class DmsCoordinateProcessor {
+  private static final Logger LOGGER = LoggerFactory.getLogger(DmsCoordinateProcessor.class);
+
+  private static final String DEGREES_LAT_REGEX_STRING = "90|[0-8]?[0-9]";
+  private static final String DEGREES_LON_REGEX_STRING = "180|1[0-7][0-9]|0?[0-9]?[0-9]";
+
+  private static final String MINUTES_REGEX_STRING = "[0-5]?[0-9]";
+  private static final String SECONDS_REGEX_STRING = "[0-5]?[0-9](?:\\.\\d*)?";
+
+  private static final String DMS_LAT_REGEX_STRING =
+      "(?:\\d+\\D|\\D)*?("
+          + DEGREES_LAT_REGEX_STRING
+          + ")\\D+("
+          + MINUTES_REGEX_STRING
+          + ")\\D+("
+          + SECONDS_REGEX_STRING
+          + ")\\D*([NnSs])\\D*";
+
+  private static final String DMS_LON_REGEX_STRING =
+      "(?:\\d+\\D|\\D)*?("
+          + DEGREES_LON_REGEX_STRING
+          + ")\\D+("
+          + MINUTES_REGEX_STRING
+          + ")\\D+("
+          + SECONDS_REGEX_STRING
+          + ")\\D*([EeWw])\\D*";
+
+  private static final int DEGREES_LAT_GROUP = 1;
+  private static final int MINUTES_LAT_GROUP = 2;
+  private static final int SECONDS_LAT_GROUP = 3;
+  private static final int DIRECTION_LAT_GROUP = 4;
+
+  private static final int DEGREES_LON_GROUP = 5;
+  private static final int MINUTES_LON_GROUP = 6;
+  private static final int SECONDS_LON_GROUP = 7;
+  private static final int DIRECTION_LON_GROUP = 8;
+
+  private static final Pattern PATTERN_DMS_COORDINATE =
+      Pattern.compile(DMS_LAT_REGEX_STRING + DMS_LON_REGEX_STRING);
+
+  // This key tells the UI that the geo is on the suggestion itself
+  private static final String LITERAL_SUGGESTION_ID = "LITERAL-DMS";
+
+  private static class CoordinateTranslator {
+
+    /**
+     * Deserialize a DMS string.
+     *
+     * @param dmsString the DMS string to deserialize.
+     * @return [dmsLat, dmsLon] The two deserialized parts of the DMS string.
+     */
+    private static List<Map<String, Object>> parseDms(final String dmsString) {
```

Comment:
❓ Could we create an inner `private static class` to encapsulate this information? Actually, we could make two: `DMSCoordinate` and `DMSComponent` such that a coordinate consists of two components, one for lat and one for long. That would remove:
1. The need for casting
2. The need for lists and to handle the semantics of N-sized data structures